addons/mail/static/src/discuss/core/common/thread_service_patch.js

```diff
@@ -47,4 +47,25 @@ patch(ThreadService.prototype, {
             thread.isLoadingAttachments = false;
         }
     },
+
+    async muteThread(thread, { minutes = false } = {}) {
+        const muteUntil = await this.rpc("/discuss/channel/mute", {
+            channel_id: thread.id,
+            minutes,
+        });
+        thread.update({
+            notification_mute_until_dt: muteUntil,
+        });
```

Comment:
Avoid calling `update`, as currently it does way too much things and there's ongoing work to remove its complexity.
Prefer immediately assigning the field with
```js
thread.mute_until_dt = muteUntil;
```